Refactor

Michiboi29 opened the pull request.

Michiboi29 committed:
folder architecture changes

ChamberlandFelix committed:
update path

ChamberlandFelix committed:
img path

Michiboi29 committed:
refactor debug

Michiboi29 committed:
update requirements toml

readme not completed

gabrielpgagne reviewed (changes requested):
Overall, je pense qu'il faudrait restructurer pour que ça aie plus de cohésion. Puis de limiter la taille du repo en enlevant les fichiers binaires (modèles, images, audio) et en faisant des helpers pour pull ces dits fichiers lorsque nécessaire.

Michiboi29 committed:
modified files structure and import fix

Michiboi29 committed:
serial reading realtime Done

missing output streamers

Michiboi29 committed:
waiting_data_implicit in read function

follows more the Interface

Michiboi29 committed:
realtime models outputs

subscriber output callback you can register to have the outputs update
in realtime

Michiboi29 committed:
update versions

Michiboi29 committed:
colormatrix WIP

Michiboi29 committed:
colormatrix WIP

Michiboi29 committed:
Merge branch 'refactor' of github.com:SBIOML/emager-py into refactor

Michiboi29 committed:
Merge branch 'refactor' of github.com:SBIOML/emager-py into refactor

Michiboi29 committed:
Merge branch 'refactor' of github.com:SBIOML/emager-py into refactor

Michiboi29 committed:
data record in progress

gabrielpgagne reviewed (changes requested):
Au moment de merge, je propose aussi que tous les if __name__ == "__main__", on delete ça et ceux pertinents on les met dans les exemples

Michiboi29 commented:
je vais delete les fichiers inutiles et les if==main aussi et j'ai déjà plusieurs exemple que je trouve pertinant si tu pense a d'autre que tu aimerais rajouter tu peux me le dire si tu veux que j'en fasse un de plus. 
Quelques points précis a regarder:

1. J'ai complété realtime_prediction avec la lib emager_py
2. J'ai changé la fonction read de Serial streamer pour que le real time sois fonctionnel, mais il faudrait peut-etre plus que une fonction read. J'ai rajouter un wait avant la lecture pour que le thread de lecture ne prenne pas tout le CPU. Sinon le realtime crash. Je pensais aussi rajouter une read avec un temps specifique.
3. J'ai modifier screen_guided_trainning pour plus ressembler au data_record et new_recording (WIP)

Michiboi29 committed:
select images Done

Michiboi29 committed:
fix save data bug

Michiboi29 committed:
selection images fix

Michiboi29 committed:
data generator bug fix

the generator is now a background thread

Michiboi29 committed:
remove deprecated files and if=main

added exemples for each if=main and more

Michiboi29 committed:
thread kill with event

Michiboi29 committed:
small cleanup

Michiboi29 commented:
@gabrielpgagne j'ai pas mal fini j'entendrais juste ton review pour commencer à merge.

gabrielpgagne commented on emager_py/realtime_prediction.py:
```diff
@@ -0,0 +1,198 @@
+    return b_combined, a_combined, zi
+
+
+class HDEMG(object):
```
Est-ce que c'est un oubli de HDEMG?

gabrielpgagne reviewed (commented):
Avant que j'accepte de merge faudrait juste:
- git merge main into refactor (pour le module`torch/`) pis cleaner les `if __name__ == "__main__` de `torch`
- je renommerais data_processings à data, visualisation à visuali**z**ation, exemples à ex**a**mples, puis mettre tous les fichiers en snake_case (realtime_GUI -> realtime_gui), idem pour les exemples

Michiboi29 committed:
Refactor file name and imports for consistency

Michiboi29 committed:
Merge branch 'main' into refactor

Michiboi29 committed:
examples fix

Michiboi29 commented:
@gabrielpgagne 
Les examples   
get_tripple_data.py , redis_remote_operations.py, redis_streamer.py, torch_models.py and virtual_tcp_streamer.py 
ne fonctionne pas

aussi build_dataflow.py je ne pense pas qu'il est à jour et je ne suis pas sur de ce qu'il devrais faire.

Je ne sais pas c'est quoi ton setup pour c'est examples la, mais il faudrait l'indiqué.

Michiboi29 commented:
@gabrielpgagne are we good to merge ?

gabrielpgagne commented:
> @gabrielpgagne Les examples get_tripple_data.py , redis_remote_operations.py, redis_streamer.py, torch_models.py and virtual_tcp_streamer.py ne fonctionne pas
> 
> aussi build_dataflow.py je ne pense pas qu'il est à jour et je ne suis pas sur de ce qu'il devrais faire.
> 
> Je ne sais pas c'est quoi ton setup pour c'est examples la, mais il faudrait l'indiqué.

Il semble que ces 4 exemples utilisent Redis. Si t'as pas un serveur Redis qui tourne sur ton ordi c'est sûr que ça marchera pas, j'vais aller mettre une note dans les scripts.

Build dataflow est outdated, mais c'est spécifique pour compiler un modèle pour FPGA so jusqu'à temps que Emager V2 soit up and running c'est correct.

gabrielpgagne committed:
minor bugfixes, fix examples

gabrielpgagne commented on pyproject.toml:
```diff
@@ -12,10 +12,13 @@ dependencies = [
     "pillow>=10.3.0",
     "sv-ttk>=2.6.0",
+    "tensorflow>= 2.15",
+    "tensorflow-io-gcs-filesystem==0.31.0",
```
C'est quoi cette dépendance là? Elle brise `pdm update`

Michiboi29 replied:
ouin ça dependant de l'ordi ou pte la version de python que j'utilise j'ai pas le choix de l'avoir je suis pas certain pourquoi. J'ai pas tester avec tous les ordi mais sur le miens j'ai pas le choix de l'avoir sinon le pdm update marche pas

Michiboi29 replied:
je vais réessayé avec juste   "tensorflow>= 2.15", voir si je peux arrivé a enlevé l'autre dependances

Michiboi29 replied:
@gabrielpgagne ok je sais pas trop ce qui sais passé, masi maintenant ça marche sans   "tensorflow-io-gcs-filesystem==0.31.0", donc on peut l'enlevé i guess.

gabrielpgagne reviewed (approved):
LGTM après ce comment là!

gabrielpgagne committed:
remove weird dependency

gabrielpgagne commented:
@Michiboi29 je te laisse merge ça si t'es prêt!
